docs/sources/userguide/plugins.md

```diff
@@ -0,0 +1,157 @@
+Docker Plugins MVP
+==================
+
+Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
+
+HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
+
+Plugins are distributed as containers.
+
+This is targeted at Docker 1.7.
+
+# Usage
+
+Plugins can be loaded using a special `docker plugin load` command, as follows:
```

Comment:
What is an agent?

On Friday, March 6, 2015, Jeff Lindsay <notifications@github.com> wrote:

> In docs/sources/userguide/plugins.md
> <https://github.com/cpuguy83/docker/pull/3#discussion_r25986848>:
>
> > @@ -0,0 +1,157 @@
> > +Docker Plugins MVP
> > +==================
> > +
> > +Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
> > +
> > +HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
> > +
> > +Plugins are distributed as containers.
> > +
> > +This is targeted at Docker 1.7.
> > +
> > +# Usage
> > +
> > +Plugins can be loaded using a special `docker plugin load` command, as follows:
>
> After discussing general usefulness of hidden/system/agent containers and
> that they would pretty much all operate with the same semantics of plugins,
> maybe we can call this subcommand agent? To then decouple this (also so
> it can be a standalone PR) from plugins, we can make the plugin handshake
> protocol optional with a flag. Another flag would provide the Docker socket
> as discussed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cpuguy83/docker/pull/3/files#r25986848>.
>